week-1-ts-ecosystem/w01e02-enter-the-world-of-typescript/src/framework/dom-creators.ts

```diff
@@ -0,0 +1,15 @@
+type tagName = 'p' | 'div' | 'section' | 'article';
+
+const createElement = (tagName: tagName = 'div', className: string = ''): HTMLElement => {
```

Comment:
Z góry zakładasz że zawsze będzie zwarcane `HTMLElement` to nie do końca prawda - można to zrobić bardziej elastycznie traktując `createElement` jako funkcję generyczną która rozszerza typ `TagName` - pokombinuj.

Wtedy będziesz miał dokładne typowanie dla `p()` / `div()` etc.... tam poniżej